java/cdk-custom-resources/src/test/java/sleeper/cdk/custom/TableDefinerLambdaIT.java

```diff
@@ -114,7 +149,212 @@ public void shouldCreateTableWithMultipleSplitPoints() throws IOException {
                         .buildList());
     }
 
+    @Test
+    public void shouldFailToDeleteTableThatDoesNotExist() {
+        // Given
+        HashMap<String, Object> resourceProperties = new HashMap<>();
+        resourceProperties.put("tableProperties", tableProperties.saveAsString());
+
+        CloudFormationCustomResourceEvent event = CloudFormationCustomResourceEvent.builder()
+                .withRequestType("Delete")
+                .withResourceProperties(resourceProperties)
+                .build();
+
+        // When / Then
+        assertThatThrownBy(() -> tableDefinerLambda.handleEvent(event, null))
+                .isInstanceOf(TableNotFoundException.class);
+
+    }
+
+    @Test
+    void shouldTakeTableOfflineWhenDeleteCalledAndInstancePropertySetFalse() throws Exception {
```

Comment:
The instance property is true, not false. It might help to clarify it's set to retain table data.